exploration/selection-declaration.md

```diff
@@ -182,6 +182,42 @@ Examples:
 - Can produce a mismatch between formatting and selection, since the operand's formatting
   isn't visible to the selector.
 
+### Require annotation of selector variables in placeholders
+
+In this alternative, the pre-existing validity requirement
+
+> Each _selector_ MUST have an _annotation_,
+> or contain a _variable_ that directly or indirectly references a _declaration_ with an _annotation_.
+
+is expanded to also require later uses of a variable that's used as a selector to be annotated:
+
+> In a _complex message_,
+> each _placeholder_ _expression_ using the same _operand_ as a _selector_ MUST have an _annotation_,
+> or contain a _variable_ that directly or indirectly references a _declaration_ with an _annotation_.
+
+Example valid message:
+```
+.input {$n :integer}
+.match {$n :number minimumFractionDigits=2}
+* {{Formats '$n' as an integer: {$n}}}
+```
```

Comment:
The above is not only bizarre, but also error prone. I am dead-set against this.

A basic goal, set out a while ago, is to prevent errors like this one directly causes, where the plural category of the match differs from the plural category of the formatted number.

Reply:
I copied this example from the "Do Nothing" alternative, and I agree that it's bad practice, even if it is technically valid. This alternative does not make all bad ideas impossible, but at least it makes this invalid:
```
.match {$n :number minimumFractionDigits=2}
* {{Error due to missing annotation: {$n}}}
```
and that is the far more common error.

Reply:
I also prefer other other alternatives over this, but I don't understand what's bizarre about the example. It's completely logical.

```javascript
n = Math.trunc(n);
switch (Intl.NumberFormat(locale, { minimumFractionDigits: 2 }).format(n)) {
  default:
  return `Format '$n' as an integer: ${n}`
}
```

No one would expect `n` in that template to be the formatted number. That said, I agree that for whatever reason it will result in some errors. I'm just not sure avoiding the option over documenting and educating is the right answer. If it's deemed the best option otherwise, of course.

Reply:
This is a toy example. A real one would have plural categories and possibly literals; and that's where the selection behavior matters.

Reply:
Not really a hill worth dying on but I don't see how that changes anything. Perhaps you could expand on "bizarre".